specs/configuration_files_specification.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Configuration Files Specification
+layout: posts
+version: 1
+---
+
+# Configuration Files Specification
+
+## Introduction
+
+Various specifications attempt to define configuration files and file formats. This
+specification establishes where these files should be looked for, in which
+order, and how precedence, masking, extensions and overrides work.
+
+The purpose of the rules defined here is to allow OS vendors to implement the
+hermetic-usr pattern, where all vendor files are shipped in the vendor tree itself
+(`/usr/`), including configuration files with system defaults, while allowing local
+or vendor overrides without modifying the original files, for easier management. This is
+especially beneficial for image-based deployments, where the vendor tree is read-only.
+
+These rules are derived from existing real-world usage from the [systemd
+project](https://github.com/systemd/systemd) and the [libeconf
+project](https://github.com/openSUSE/libeconf).
+
```

Comment:
hmm, I think this should focus on two distinct cases:

1. the case where a single object (for example a daemon) is configured via a main config file for that single object + drop-ins. example: logind.conf for systemd-logind.

2. the case where multiple objects are configured via only-drops without any "main config file". example: tmpfiles.d/ sysusers.d/ and such.

The semantics of bot are of course very similar, but the absence of the "main config" file in the latter case, is important to mention.